p2p/net/udp.go

```diff
@@ -413,6 +412,10 @@ func (ucw *udpConnWrapper) Write(b []byte) (int, error) {
 }
 
 func (ucw *udpConnWrapper) Close() error {
-	close(ucw.closeChan)
+	select {
```

Comment:
this is to protect double close I assume- who might double closes a connection?

Reply:
udpConnWrapper can be closed when any of the goroutines created by MsgConnection will exit, or when it is evicted by UDPNet. in the second case Close will be called directly on udpConnWrapper. i thought it is fine to double-close, but i think I can modify the eviction code to call Close on MsgConnection instead if you prefer

previously it was only closed when evicted, but I want to avoid any dangling goroutines to simplify debugging next time